oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/BasicDocumentStoreTest.java

```diff
@@ -54,6 +55,18 @@ public BasicDocumentStoreTest(DocumentStoreFixture dsf) {
         super(dsf);
     }
 
+    @After
+    public void tearDown() {
+        Revision.resetClockToDefault();
+        markDocumentsForCleanup();
+    }
+
+    private void markDocumentsForCleanup() {
+        for (NodeDocument doc : Utils.getAllDocuments(ds)) {
+            removeMe.add(doc.getId());
+        }
+    }
+
```

Comment:
Why is this needed in BasicDocumentStoreTest? AFAICS none of the tests uses a Clock. I think this should rather go to `MissingLastRevSeekerTest` and clean up documents created by the new `getNonSplitDocs()` test.

Reply:
There was one node created by this class which was not getting cleaned up and causing the newly created test to fail. Thats why I added this cleanup over here. I will dig deeper in this.

Reply:
Basically this [0] is the culprit function creating a node but not able to delete it. Need to either fix this or can add after cleanup function as already done.

[0] : https://github.com/apache/jackrabbit-oak/blob/trunk/oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/BasicDocumentStoreTest.java#L1144

Reply:
Fixed the culprit test case from BasicDocumentStoreTest class